spec/reports/uncommitted_holdings_spec.rb

```diff
@@ -0,0 +1,146 @@
+# frozen_string_literal: true
+
+require "clustering/cluster_commitment"
+require "reports/uncommitted_holdings"
+require "spec_helper"
+require_relative "../../bin/reports/compile_uncommitted_holdings"
+
+RSpec.describe Reports::UncommittedHoldings do
+  let(:report_all) { described_class.new({all: true}) }
+
+  let(:ocn1) { 5 }
+  let(:org1) { "umich" }
+  let(:loc1) { "i123" }
+  let(:ht_spm1) { build(:ht_item, :spm, ocns: [ocn1]) }
+  let(:hol1) { build(:holding, ocn: ocn1, local_id: loc1, organization: org1) }
+  let(:com1) { build(:commitment, ocn: ocn1, local_id: loc1, organization: org1) }
+
+  let(:ocn2) { 6 }
+  let(:org2) { "smu" }
+  let(:loc2) { "i456" }
+  let(:ht_spm2) { build(:ht_item, :spm, ocns: [ocn2]) }
+  let(:hol2) { build(:holding, ocn: ocn2, local_id: loc2, organization: org2) }
+  let(:com2) { build(:commitment, ocn: ocn2, local_id: loc2, organization: org2) }
+
+  let(:mpm) { build(:ht_item, :mpm, ocns: [ocn1]) }
+  let(:ser) { build(:ht_item, :ser, ocns: [ocn1]) }
+
+  # Clusters and saves each element in an array of clusterables
+  def cluster_tap_save(clusterables)
+    clusterables.each do |clusterable|
+      case clusterable.class
+      when Clusterable::Holding
```

Comment:
I think this approach might be helpful to apply elsewhere in the tests. (It also makes it tempting to apply some kind of Rails-style metaprogramming to the `Clusterable` class name to get the corresponding `Clustering` class, which I think is best avoided even when straightforward...)

It's also worth noting that we don't really ever need this correspondence in the code itself (it really only shows up in the `Loader` classes, which know about the corresponding `Clusterable` and `Clustering` classes), but it shows up over and over in setup for the specs.

Reply:
Moved.